docs/container-based-setup.adoc

```diff
@@ -0,0 +1,113 @@
+
+Start the Storage node or Client as shown in the example below.
+
+----
+sudo docker run -d                            \
+    --network host                            \
+    -v /root/kadalu/workdir:/var/lib/kadalu   \
+    -v /root/kadalu/home:/root/.kadalu        \
+    -v /ZP1/storage1:/ZP/storage1             \
+    -v /sys/fs/cgroup/:/sys/fs/cgroup:ro      \
+    --privileged                              \
+    --name kadalu-$(hostname)                 \
+    --hostname $(hostname)                    \
+    kadalu/storage-node:latest
+----
+
+== Storage Units
+
+----
+..
+    -v /ZP1/storage1:/ZP/storage1             \
+..
+----
+
+Pre create the file system(xfs/ext4/zfs) and expose the Storage while starting the Kadalu Storage container. Provide multiple Storage units per container/node as required. For example, the following code shows creating ZFS dataset and exposing that while starting the Storage node.
+
+----
+sudo zpool create ZP1 /dev/vda
+sudo zfs create ZP1/storage1
+----
+
+Storage from host are exposed to the Storage container.
+
+Storage Units are available as Volumes inside container, use those to Create Kadalu Volumes.
+
+**Note**: This approach will not work in Mac based container setup since these Volumes are not exposed directly from the host machine.
+
+== Other Persistent Volumes
+
+----
+..
+    -v /root/kadalu/workdir:/var/lib/kadalu   \
+    -v /root/kadalu/home:/root/.kadalu        \
+..
+----
+
+Along with Storage units a few other directories needs to be persisted.
+
+- `/var/lib/kadalu` - If a Container becomes Mgr instance then it stores full Cluster configurations, else if it is Agent instance, it will contain the tokens that are required to communicate with Mgr instances. 
+- `/root/.kadalu` - This is optional, but good to have this persisted to avoid running login command every time on container restart.
+
+== Network - Host
+
+----
+..
+    --network host                            \
+..
+----
+
+Kadalu Storage nodes will communicate with each other, prefered to use Host network option while starting the Storage container. A few more fixes required to make the Storage manager to customize the Port. Once it gets that capability, possibly other network modes can be supported.
+
+**Note**: Since network=host option is in use, starting multiple Storage containers per node is not possible. Instead expose multiple Storage units per instance of Storage container.
```

Comment:
wanted to have more understanding on this. But if you are aware of what the issues are, good to go ahead with this version.

Reply:
When multiple Storage manager started in same node(different containers), using the same port is not possible by two containers with network=host. Without this option, different Port can be used externally but within container it can use `3000` itself (`-p 3001:3000` for one container and `-p 3002:3000` for another).

Some more changes required to join the nodes with different port and communicate. For now, `network=host` looks easy approach.